azurerm/internal/services/network/express_route_connection_resource_test.go

```diff
@@ -0,0 +1,226 @@
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func testAccExpressRouteConnection_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_express_route_connection", "test")
+	r := ExpressRouteConnectionResource{}
+
+	data.ResourceSequentialTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r ExpressRouteConnectionResource) Exists(ctx context.Context, client *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	expressRouteConnectionClient := client.Network.ExpressRouteConnectionsClient
+	id, err := parse.ExpressRouteConnectionID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := expressRouteConnectionClient.Get(ctx, id.ResourceGroup, id.ExpressRouteGatewayName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+
+		return nil, fmt.Errorf("retrieving Express Route Connection %q: %+v", state.ID, err)
+	}
+
+	return utils.Bool(resp.ExpressRouteConnectionProperties != nil), nil
+}
+
+func (r ExpressRouteConnectionResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_connection" "test" {
+  name                             = "acctest-ExpressRouteConnection-%d"
+  express_route_gateway_id         = azurerm_express_route_gateway.test.id
+  express_route_circuit_peering_id = azurerm_express_route_circuit_peering.test.id
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ExpressRouteConnectionResource) requiresImport(data acceptance.TestData) string {
+	config := r.basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_express_route_connection" "import" {
+  name                             = azurerm_express_route_connection.test.name
+  express_route_gateway_id         = azurerm_express_route_connection.test.express_route_gateway_id
+  express_route_circuit_peering_id = azurerm_express_route_connection.test.express_route_circuit_peering_id
+}
+`, config)
+}
+
+func (r ExpressRouteConnectionResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_route_table" "test" {
+  name           = "acctest-VHUBRT-%d"
+  virtual_hub_id = azurerm_virtual_hub.test.id
+}
+
+resource "azurerm_express_route_connection" "test" {
+  name                             = "acctest-ExpressRouteConnection-%d"
+  express_route_gateway_id         = azurerm_express_route_gateway.test.id
+  express_route_circuit_peering_id = azurerm_express_route_circuit_peering.test.id
+  routing_weight                   = 2
+  authorization_key                = "90f8db47-e25b-4b65-a68b-7743ced2a16b"
```

Comment:
Do we have a better way than hard-code this key here? And should we regard this as a credential?

Reply:
Here is the fake test data. So I think we don't care about it.